src/Stack/Clean.hs

```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE CPP #-}
```

Comment:
If possible, it's best to avoid adding CPP, since it can cause unnecessary recompilation. Can you instead use qualified usage of the identifier?

Alternatively: We can bump the minimum version of `rio` in the cabal file.

Alternative alternatively: we simply don't care about this, since Stack should always be built from the relevant `stack.yaml` file.

Reply:
I've caught this incompatibility when trying to build Stackage, if we just pin dependencies then we'll need to drop Stack from Stackage and this PR was thought to prevent that from happening

Reply:
OK, fair enough, I rescind that idea. But still, finding ways that don't introduce `CPP` would be ideal.